cmd/apiserver_receive_adapter/main.go

```diff
@@ -18,55 +18,96 @@ package main
 
 import (
 	"flag"
-	"log"
+	"fmt"
 
-	// Uncomment the following line to load the gcp plugin (only required to authenticate against GKE clusters).
+	// Uncomment the following line to load the gcp plugin
+	// (only required to authenticate against GKE clusters).
 	// _ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
 
 	"github.com/kelseyhightower/envconfig"
 	"go.uber.org/zap"
-	"go.uber.org/zap/zapcore"
 	"k8s.io/apimachinery/pkg/api/meta"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/client-go/dynamic"
 	"k8s.io/client-go/tools/clientcmd"
 	"knative.dev/eventing/pkg/adapter/apiserver"
 	"knative.dev/eventing/pkg/kncloudevents"
 	"knative.dev/eventing/pkg/tracing"
+	"knative.dev/eventing/pkg/utils"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/metrics"
 	"knative.dev/pkg/signals"
 )
 
+const (
+	component = "apiserversource"
+)
+
 var (
 	masterURL  = flag.String("master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
 	kubeconfig = flag.String("kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
 )
 
 type envConfig struct {
-	Namespace  string   `envconfig:"SYSTEM_NAMESPACE" default:"default"`
-	Mode       string   `envconfig:"MODE"`
-	SinkURI    string   `split_words:"true" required:"true"`
-	ApiVersion []string `split_words:"true" required:"true"`
-	Kind       []string `required:"true"`
-	Controller []bool   `required:"true"`
+	Namespace         string   `envconfig:"SYSTEM_NAMESPACE" default:"default"`
+	Mode              string   `envconfig:"MODE"`
+	SinkURI           string   `split_words:"true" required:"true"`
+	ApiVersion        []string `split_words:"true" required:"true"`
+	Kind              []string `required:"true"`
+	Controller        []bool   `required:"true"`
+	ApiServerImporter string   `envconfig:"APISERVERIMPORTER" required:"true"`
```

Comment:
maybe just `Name`?
There is still some debate on importer/source. I think it's fine to simply use `Name` here... and pass variable as NAME as well...

Reply:
changed this to NAME